services/connector/connector_flows_test.go

```diff
@@ -0,0 +1,111 @@
+		var evt commands.EventType
+		err := json.Unmarshal(s, &evt)
+		assert.NoError(t, err)
+
+		switch evt.Type {
+		case signal.EventConnectorSendRequestAccounts:
+			var ev signal.ConnectorSendRequestAccounts
+			err := json.Unmarshal(evt.Event, &ev)
+			assert.NoError(t, err)
+
+			err = api.RequestAccountsFinished(commands.RequestAccountsFinishedArgs{
+				Accounts: []types.Address{accountAddress},
+				Error:    nil,
+			})
+			assert.NoError(t, err)
+		case signal.EventConnectorSendTransaction:
+			var ev signal.ConnectorSendTransactionSignal
+			err := json.Unmarshal(evt.Event, &ev)
+			assert.NoError(t, err)
+
+			err = api.SendTransactionFinished(commands.SendTransactionFinishedArgs{
+				Hash:  expectedHash,
+				Error: nil,
+			})
+			assert.NoError(t, err)
+		}
+	}))
+
+	// Request accounts, now for real
+	request = "{\"method\": \"eth_requestAccounts\", \"params\": [], \"origin\": \"http://testDAppURL123\", \"dAppName\": \"testDAppName\", \"dAppIconUrl\": \"http://testDAppIconUrl\" }"
+	expectedResponse := strings.ToLower(fmt.Sprintf(`{"accounts":["%s"]}`, accountAddress.Hex()))
+	response, err = api.CallRPC(request)
+	assert.NoError(t, err)
+	assert.Equal(t, expectedResponse, response)
+
+	// Request to switch ethereum chain
+	expectedChainId := 0x5
+	request = fmt.Sprintf("{\"method\": \"wallet_switchEthereumChain\", \"params\": [%d], \"origin\": \"http://testDAppURL123\", \"dAppName\": \"testDAppName\", \"dAppIconUrl\": \"http://testDAppIconUrl\" }", expectedChainId)
+	expectedResponse = fmt.Sprintf(`%d`, expectedChainId)
+	response, err = api.CallRPC(request)
+	assert.NoError(t, err)
+	assert.Equal(t, expectedResponse, response)
+
+	// Check if the chain was switched
+	request = "{\"method\": \"eth_chainId\", \"params\": [], \"origin\": \"http://testDAppURL123\", \"dAppName\": \"testDAppName\", \"dAppIconUrl\": \"http://testDAppIconUrl\" }"
+	response, err = api.CallRPC(request)
+	assert.NoError(t, err)
+	assert.Equal(t, expectedResponse, response)
+
+	// Check the account after switching chain
+	request = "{\"method\": \"eth_accounts\", \"params\": [], \"origin\": \"http://testDAppURL123\", \"dAppName\": \"testDAppName\", \"dAppIconUrl\": \"http://testDAppIconUrl\" }"
+	expectedResponse = strings.ToLower(fmt.Sprintf(`{"accounts":["%s"]}`, accountAddress.Hex()))
+	response, err = api.CallRPC(request)
+	assert.NoError(t, err)
+	assert.Equal(t, expectedResponse, response)
+
+	// Send transaction
+	request = fmt.Sprintf("{\"method\": \"eth_sendTransaction\", \"params\":[{\"from\":\"%s\",\"to\":\"0x0200000000000000000000000000000000000000\",\"gas\":null,\"gasPrice\":null,\"value\":\"0x0\",\"nonce\":null,\"maxFeePerGas\":null,\"maxPriorityFeePerGas\":null,\"input\":\"0x\",\"data\":\"0x307830\",\"MultiTransactionID\":0,\"Symbol\":\"\"}], \"origin\": \"http://testDAppURL123\", \"dAppName\": \"testDAppName\", \"dAppIconUrl\": \"http://testDAppIconUrl\" }", accountAddress.Hex())
```

Comment:
Looking at this I wonder is it's escaped correctly. especially because of the `http://...`. I would expect `http:\/\/...` or something? More reason to have a function we can trust to do the escaping.

Reply:
the implementation is expecting exact this format, only double quotes should be escaped

Reply:
Surely goloang doesn't have a way to create raw strings? :thinking: 

Yeah: https://go.dev/ref/spec#raw_string_lit

Reply:
In `/commands` there are tests in which internal serialisation of go is used, and this test is rather necessary to show what raw requests are expected by the service, so it makes sense to store them in this form, so that they can be inserted into postman. For what do we need escaping, i answered a little above